test/e2e/box_record_test.go

```diff
@@ -0,0 +1,120 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func TestBoxRecording(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping testing in short mode")
+	}
+
+	// given
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	network := createNetwork(ctx, t)
+	defer network.Remove(ctx)
+
+	boxName := randomString("box-")
+
+	// when
+	box := startBoxWithEnv(ctx, t, boxName, network.name)
+	defer box.Terminate(ctx)
+
+	err := startRecordTester(ctx)
+	require.NoError(t, err)
+}
+
+func startBoxWithEnv(ctx context.Context, t *testing.T, hostname, network string) *catalystContainer {
+	req := testcontainers.ContainerRequest{
+		Image:        "livepeer/in-a-box",
+		Hostname:     hostname,
+		Name:         hostname,
+		Networks:     []string{network},
+		ExposedPorts: []string{"1935:1935/tcp", "8888:8888/tcp"},
+		ShmSize:      1000000000,
+		WaitingFor:   wait.NewLogStrategy("API server listening"),
+	}
+	container, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          false,
+	})
+	require.NoError(t, err)
+
+	// Redirect container logs to the standard logger
+	lc := logConsumer{name: hostname}
+	err = container.StartLogProducer(ctx)
+	require.NoError(t, err)
+	container.FollowOutput(&lc)
+
+	err = container.Start(ctx)
+	require.NoError(t, err)
+
+	// Store mapped ports
+	catalyst := &catalystContainer{
+		Container: container,
+		hostname:  hostname,
+	}
+
+	mappedPort, err := container.MappedPort(ctx, boxPort)
+	require.NoError(t, err)
+	catalyst.box = mappedPort.Port()
+
+	mappedPort, err = container.MappedPort(ctx, rtmpPort)
+	require.NoError(t, err)
+	catalyst.rtmp = mappedPort.Port()
+
+	// container IP
+	cid := container.GetContainerID()
+	dockerClient, err := testcontainers.NewDockerClient()
+	require.NoError(t, err)
+	inspect, err := dockerClient.ContainerInspect(ctx, cid)
+	require.NoError(t, err)
+	catalyst.ip = inspect.NetworkSettings.Networks[network].IPAddress
+
+	return catalyst
+}
+
+func startRecordTester(ctx context.Context) error {
+	err := run(
+		ctx,
+		"go",
+		"run",
+		"github.com/livepeer/stream-tester/cmd/recordtester",
```

Comment:
Don't we want to run a dockerized record-tester? I think it would play better with all the test containers and networking.

Reply:
That would definitely be an improvement but there are lots of restrictions with the box right now where things are hardcoded to `127.0.0.1`, such as the redirect URLs from the catalyst-api load balancing. I tried and failed to get a containerized version working with host networking, resulting in [this bug](https://github.com/testcontainers/testcontainers-go/issues/1514). This localhost restriction will be removed with [SRE-159](https://linear.app/livepeer/issue/SRE-159/livepeer-in-a-box-allow-customization-of-hostnameip-on-boot) and we can revisit.